### ISSUE TITLE
Evaluation of model not working due to keras compile error

### ISSUE THREAD

**Issue by bjornmosten** [OlafenwaMoses/ImageAI#464]
Using the example code provided, this is the error given when trying to evaluate a model:
`Using TensorFlow backend.
Starting Model evaluation....
^Ti/home/user/.local/lib/python3.6/site-packages/keras/engine/saving.py:341: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '
skipping the evaluation of ../imageai/models/detection_model-ex-029--loss-0008.378.h5 because following exception occurred: 'NoneType' object has no attribute 'shape'`

**Comment by rola93** [comment]
Are you sure your dataset is properly saved on the right directories? I guess the problem is either in your dataset location or in your .xml file format. Double check those points, please.

In adition to this: can you share the full traceback? Can you share your code?

Thanks

**Comment by bjornmosten** [comment]
Yes, I have tried putting the datasets in all different kinds of locations. I have also tried using absolute paths. Considering training and the model itself works just fine with detection and all. I don't understand what the problem would be with the XML files (they are, however, automatically produced). I attached one of the XML files (in txt format). 
The code I used was the one given by the repo in _examples/custom_detection_single_model_evaluation.py_
`
from imageai.Detection.Custom import DetectionModelTrainer

trainer = DetectionModelTrainer()
trainer.setModelTypeAsYOLOv3()
trainer.setDataDirectory(data_directory="hololens")
trainer.evaluateModel(model_path="hololens-ex-60--loss-2.76.h5", json_path="detection_config.json", iou_threshold=0.5, object_threshold=0.3, nms_threshold=0.5)

# download JSON file via https://github.com/OlafenwaMoses/ImageAI/releases/download/essential-v4/detection_config.json
# download detection model via https://github.com/OlafenwaMoses/ImageAI/releases/download/essential-v4/hololens-ex-60--loss-2.76.h5



"""
SAMPLE RESULT
Model File:  hololens_detection_model-ex-09--loss-4.01.h5 
Using IoU :  0.5
Using Object Threshold :  0.3
Using Non-Maximum Suppression :  0.5
hololens: 0.9613
mAP: 0.9613
===============================
"""
`

[1579944741.9134734.txt](https://github.com/OlafenwaMoses/ImageAI/files/4455547/1579944741.9134734.txt)

**Comment by LeoBel0799** [comment]
I have the same problem
